conda/recipes/libkvikio/meta.yaml

```
@@ -75,13 +94,21 @@ outputs:
     build:
       number: {{ GIT_DESCRIBE_NUMBER }}
       string: cuda{{ cuda_major }}_{{ date_string }}_{{ GIT_DESCRIBE_HASH }}_{{ GIT_DESCRIBE_NUMBER }}
+      ignore_run_exports_from:
+        {% if cuda_major == "11" %}
+        - {{ compiler('cuda11') }}
+        {% endif %}
     requirements:
       build:
         - cmake {{ cmake_version }}
       run:
```

This is missing DSOs, like we had for librmm-tests.
```
WARNING (libkvikio-tests,bin/tests/libkvikio/BASIC_IO_TEST): $RPATH/libcudart.so.12 not found in packages, sysroot(s) nor the missing_dso_whitelist.
.. is this binary repackaging?
```
For that package, we solved it with:
```suggestion
      host:
        - cuda-version ={{ cuda_version }}
        {% if cuda_major == "11" %}
        - cudatoolkit ={{ cuda_version }}
        {% else %}
        - cuda-cudart-dev
        {% endif %}
      run:
```
Then use `pin_compatible` in `run:`.

But if we're taking the approach of putting `cuda-cudart-dev` in `host` in order to get its `run_export` of `cuda-cudart` into `run`, should we do the same with `libcufile` and list its dev package in host instead of run? 😕 Figuring out how to be consistent here is important to set an example for the rest of RAPIDS.

---

Let's make the change proposed here

Would file [an issue on the `libcufile` feedstock]( https://github.com/conda-forge/libcufile-feedstock/issues/new ) for any follow up work there

---

There's no work needed on `libcufile` here, since [the `-dev` package already has a run-export](https://github.com/conda-forge/libcufile-feedstock/blob/5a47af81743d9edfc60b1acf29b79c76fd184b5c/recipe/meta.yaml#L75-L76). My question is about how we should specify RAPIDS dependencies.

We saw that librmm-tests was missing a DSO for `libcudart.so.12`. That is provided by `cuda-cudart`. [John proposed](https://github.com/rapidsai/rmm/pull/1223#pullrequestreview-1489881267) to 
have `host: cuda-cudart-dev` to fix this missing DSO _rather than_ `run: cuda-cudart` as I expected. We merged that change.

Do we think it's better to list CUDA libraries (including `cuda-cudart` and `libcufile`) as "dev" packages in the `host`, rather than just listing the actual runtime library in `run`? I don't necessarily see why this is preferred. I wrote a longer comment here: https://github.com/rapidsai/rmm/pull/1223#issuecomment-1599985240 which proposed two other solutions (putting the "dev" package in `run` for the header-only parent library, or just listing the non-dev package in `run` for anything that is overlinking). Let's discuss.

---

Yes if we are building against / linking to these libraries at some point (as is the case with `libkvikio-tests`), then we should have them in `host` for that package (`libkvikio-tests` here)

---

@jakirkham The issue here is that we have a split recipe. We have `-dev` packages in the top-level host. Should we be re-listing these in `host` for each of the package outputs? I think it would be sufficient to list `-dev` packages in the top-level `host` and then only list the runtime libraries in the `run` of the package output. Does this not work? Currently we have no `host:` section for most package outputs, only the top level.

---

Conclusion: conda-build won't correctly detect linked DSOs from the `run:` section of a split package output. We decided to use `host:` here.

See also: https://github.com/conda/conda-build/issues/3478.

In an ideal world, we'd be able to get the `run_exports` from the top-level into the `run:` dependencies of the subpackage outputs, and the DSO link check would be able to see the top-level host/run environment contains the desired DSOs.

---

Resolved in 4d8fd6e / fbcb1b0.